### PR TITLE
Update Ubuntu Bionic to Focal in setup.rst

### PR DESCRIPTION
**Pull request by illia-v**
Ubuntu Focal Fossa is the latest LTS release. Its default Python version is 3.8.

Also, I added a numeric version of the Ubuntu release because it is likely more familiar to users.

### CODE REVIEW

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
@illia-v Looks good, please could you resolve the conflicts?

File: setup.rst [hunk 1]
```diff
@@ -323,9 +323,9 @@ dependencies for the Python you're working on by using the ``apt`` command.
 First, make sure you have enabled the source packages in the sources list.
 You can do this by adding the location of the source packages, including
 URL, distribution name and component name, to ``/etc/apt/sources.list``.
-Take Ubuntu Bionic for example::
+Take Ubuntu 20.04 LTS (Focal Fossa) for example::
```
Comment (ezio-melotti) [inline setup.rst:326]:
```suggestion
Take Ubuntu 22.04 LTS (Jammy Jellyfish) for example::
```

File: setup.rst [hunk 2]
```diff
@@ -323,9 +323,9 @@ dependencies for the Python you're working on by using the ``apt`` command.
...
 
-   deb-src http://archive.ubuntu.com/ubuntu/ bionic main
+   deb-src http://archive.ubuntu.com/ubuntu/ focal main
```
Comment (ezio-melotti) [inline setup.rst:328]:
```suggestion
   deb-src http://archive.ubuntu.com/ubuntu/ jammy main
```

File: setup.rst [hunk 3]
```diff
@@ -336,7 +336,7 @@ Then you should update the packages index::
 
 Now you can install the build dependencies via ``apt``::
 
-   $ sudo apt-get build-dep python3.6
+   $ sudo apt-get build-dep python3.8
```
Comment (ezio-melotti) [inline setup.rst:339]:
```suggestion
   $ sudo apt-get build-dep python3.10
```

**Review by ezio-melotti — CHANGES_REQUESTED** [review]

**Review by ezio-melotti — APPROVED** [review]

**Pull request merged**